crates/fs/src/fs.rs

```diff
@@ -876,6 +876,7 @@ impl FakeFsState {
                         match prefix {
                             Some(prefix_component) => {
                                 canonical_path.push(prefix_component.as_os_str());
+                                canonical_path.push("/");
```

Comment:
Curious why this is needed.

Reply:
this because `prefix_component.as_os_str()` return `c:`, which is not represent path `c:/`

Reply:
Which means we have to actually `join` here instead of `push`?

Can we do that? I am somewhat confused and concerned about `/` not being a proper separator in Windows, but used in windows-only code (as `Component::Prefix` seems not to be a Unix thing).

Reply:
for windows platform `c:\\root` path using `components` will generate `[prefix, rootdir, normal]`, `prefix` only consume `c:`,  and `\\` will consume as `rootdir`. so we should `push ("/")` here.

Reply:
I did not understand, we cannot `join` here? Why?

Reply:
sorry, I misunderstand your mean, i say using `push` is also ok.
we can using `canonical_path = canonical_path.join("/");` instead of `canonical_path.push("/")`

Reply:
You also seem to completely miss the point that 

> I am somewhat confused and concerned about / not being a proper separator in Windows, but used in Windows-only code (as Component::Prefix seems not to be a Unix thing).

and insist on adding `/` for some reason there.

If we can `join,` we can join everywhere around where `canonical_path.join` is used.
And then, `canonical_path.join("/")` is noop (even with a proper separator), and we can safely `.join` all segments.

Reply:
Or are we both wrong and it's impossible to join?
https://rust-lang.github.io/rust-clippy/master/index.html#join_absolute_paths

Reply:
yeah, the clippy don't allow `join("/")` ...., I should using `push("/")` here.